src/app/component/key/key-interface.js

```diff
@@ -0,0 +1,294 @@
+          KeyInterface.pendedKeys.splice(KeyInterface.pendedKeys.indexOf(this), 1);
+        } else {
+          KeyInterface.pendedKeys.push(this);
+        }
+      }
+
+      if (this.keyType === 'spec') {
+        if (this.keyActive) {
+          KeyInterface.setFlag(this, true);
+        } else {
+          KeyInterface.setFlag(this, false);
+          if (KeyInterface.pendedKeys.length > 0) { KeyInterface.setFlagFromPended(); }
+        }
+
+        this.caserCheck();
+      }
+    }
+
+    this.typer();
+  }
+
+  press() {
+    if (!this.options || !this.options.includes('sticky')) {
+      this.setActiveState(true);
+    } else if (this.keyId === 'CapsLock') {
+      this.toggleActiveState();
+    }
+
+    if (this.keyType === 'spec') {
+      if (!this.options || !this.options.includes('sticky')) {
+        KeyInterface.setFlag(this, true);
+      } else if (this.keyActive) {
+        KeyInterface.setFlag(this, true);
+      } else {
+        KeyInterface.setFlag(this, false);
+        if (KeyInterface.pendedKeys.length > 0) { KeyInterface.setFlagFromPended(); }
+      }
+
+      this.caserCheck();
+    }
+
+    this.typer();
+  }
+
+  relese() {
+    if (this.keyId !== 'CapsLock') {
+      this.setActiveState(false);
+    } else if (this.options && this.options.includes('semisticky') && KeyInterface.pendedKeys.includes(this)) {
+      KeyInterface.pendedKeys.splice(KeyInterface.pendedKeys.indexOf(this), 1);
+    }
+
+    if (this.keyType === 'spec' && !(this.options && this.options.includes('sticky'))) {
+      KeyInterface.setFlag(this, false);
+      if (KeyInterface.pendedKeys.length > 0) { KeyInterface.setFlagFromPended(); }
+    }
+
+    if (this.keyType === 'spec') {
+      this.caserCheck();
+    }
+  }
+
+  typer() {
+    if (this.keyType === 'spec') {
+      switch (this.keyId) {
+        case 'Space': KeyInterface.typerInt.typer({ command: 'typing', value: ' ' }); break;
+        case 'Enter': KeyInterface.typerInt.typer({ command: 'typing', value: '\n' }); break;
+        case 'Tab': KeyInterface.typerInt.typer({ command: 'typing', value: '\t' }); break;
+        case 'Backspace': KeyInterface.typerInt.typer({ command: 'backspace', value: '' }); break;
+        case 'Delete': KeyInterface.typerInt.typer({ command: 'delete', value: '' }); break;
+        case 'ArrowLeft': KeyInterface.typerInt.typer({ command: 'left', value: '' }); break;
+        case 'ArrowRight': KeyInterface.typerInt.typer({ command: 'right', value: '' }); break;
+        default: break;
+      }
+
+      if (
+        (this.keyId === 'AltLeft' || this.keyId === 'AltRight'
+        || this.keyId === 'ControlLeft' || this.keyId === 'ControlRight')
+        && KeyInterface.flags.alt && KeyInterface.flags.ctrl && !KeyInterface.flags.shift
+      ) {
+        KeyInterface.nextGlobalLang();
+        KeyInterface.resetPended();
+        KeyInterface.setGlobalShellStyle();
+        KeyInterface.setGlobalChar();
```

Comment:
Those methods doesn't do anything, those are looks like procedure. Look at the concept of `pure function`.
https://en.wikipedia.org/wiki/Pure_function